Desktop/data/datasetpackage.cpp

```diff
@@ -1634,16 +1634,17 @@ void DataSetPackage::setColumnName(size_t columnIndex, const std::string & newNa
 
 	std::string oldName = getColumnName(columnIndex);
 
-	if(resetModel)
-		beginResetModel();
+	beginResetModel();
 
-	column->setName(newName);
+	bool change = column->setName(newName);
 
-	if(resetModel)
-		endResetModel();
+	endResetModel();
 
-	setManualEdits(true);
-	emit datasetChanged({}, {}, QMap<QString, QString>({{tq(oldName), tq(newName)}}), false, false);
+	if (change)
+	{
+		setManualEdits(true);
+		emit datasetChanged({}, {}, QMap<QString, QString>({{tq(oldName), tq(newName)}}), false, false);
```

Comment:
You now still give an `changed name from X to ""` signal no?
As in, if change == false because the new name is "" the new name is not newName...

Reply:
Im not saying it doesnt make sense, im saying that hes sending a signal `dataSetChanged({}, {}, "someName", "")` despite the name *NOT* changing

Reply:
No: if newName is empty, change is false, and datasetChanged signal is not emitted.